presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -381,6 +384,19 @@ private static URI getHttpUri(ServiceDescriptor descriptor, boolean httpsRequire
         return null;
     }
 
+    private static OptionalInt getThriftServerPort(ServiceDescriptor descriptor)
+    {
+        String port = descriptor.getProperties().get("thriftServerPort");
+        if (port != null) {
+            try {
+                return OptionalInt.of(Integer.parseInt(port));
+            }
+            catch (IllegalArgumentException ignored) {
```

Comment:
Looks like this `thriftServerPort` is added to `Properties` by our code and it is generated from an `int`. Is this catch still necessary (I'm assuming you are trying to catch exceptions where port could not be parsed into an integer).